src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -0,0 +1,1089 @@
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV_MAIN = 
+R"(
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = irr_glsl_pseudoMul4x4with3x1(CamData.params.MVP, vPos);
+    ViewPos = irr_glsl_pseudoMul3x4with3x1(CamData.params.MV, vPos);
+    mat3 normalMat = irr_glsl_SBasicViewParameters_GetNormalMat(CamData.params);
+    Normal = normalMat*normalize(vNormal);
+}
+)";
+    constexpr const char* FRAG_SHADER_NO_UV_MAIN =
+R"(#define Ia 0.1
+void main()
+{
+    vec3 N = normalize(Normal);
+    vec3 L = normalize(-ViewPos);
+    vec3 R = -reflect(L,N);
+    float NdotL = max(dot(N,L), 0.0);
+    float VdotR = max(dot(L,R), 0.0);
+
+    vec3 color;
+    if (PC.params.extra&ILLUM_MODEL_MASK > 0)
+    {
+        color = PC.params.Ka*Ia + PC.params.Kd*NdotL;
+        switch (PC.params.extra&ILLUM_MODEL_MASK)
+        {
+        case 2:
+        case 3://2 + reflection map
+        case 4://3 with transparency (glass)
+        case 6:
+        case 8://reflection map
+        case 9://reflection map
+            color += PC.params.Ks*pow(VdotR, PC.params.Ns);
```

Comment:
proper normalization for Blinn Phong (and Blinn-Phong is the power of NdotH, no reflected light taken into account) is:

![image](https://user-images.githubusercontent.com/6894321/72722551-e5783580-3b7e-11ea-90bf-1d259a190ade.png)

(the thing in the middle of the inequality is the actual factor)

Reply:
n is the specular power, IIRC